packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -439,6 +439,10 @@ export default class RouterService extends Service {
     }
     ```
 
+    The exact timing to fire `routeDidChange` is after the Route's
+    [didTransition](/ember/release/classes/Route/events/didTransition?anchor=didTransition)
+    action fired.
```

Comment:
I think that it is good to explicitly state this, but I'd prefer to tweak the verbiage here a bit. Maybe something:

```
`routeDidChange` will be called after any `Route`'s [didTransition](/ember/release/classes/Route/events/didTransition?anchor=didTransition) action has been fired.
```